Lib/fontmake/font_project.py

```diff
@@ -185,16 +188,15 @@ def run_from_designspace(self, designspace_path, **kwargs):
         self.run_from_ufos(ufos, **kwargs)
 
     def run_from_ufos(
-            self, ufos, compatible=False, remove_overlaps=True, mti_source=None,
-            **kwargs):
+            self, ufos, remove_overlaps=True, mti_source=None, **kwargs):
         """Run toolchain from UFO sources to OpenType binaries."""
 
         if isinstance(ufos, str):
             ufos = glob.glob(ufos)
         if isinstance(ufos[0], str):
             ufos = [Font(ufo) for ufo in ufos]
 
-        if remove_overlaps and not compatible:
+        if remove_overlaps and not self.compatible:
```

Comment:
This check does prevent overlap removal when compatibility is turned on. But we can make it fail instead of silently doing this.